### DISCUSSION TITLE
Water temperature

### DISCUSSION THREAD

**Discussion by disforw** [kdeyev/eyeonwater#53] [Ideas]
I seem to remember seeing water temperature somewhere in the old api response (before the library move) but don’t see it now. Is it possible? Am I mistaken?
I’d LOVE to have a water temperature sensor entity!!

**Comment by kdeyev** [comment]
Probably here: https://github.com/kdeyev/pyonwater/blob/6680991865968fc0b04611f33827eda4800a7911/pyonwater/models/eow_models.py#L334C17-L334C17
I built these data models based on the response I received from EOW, but it may easily happen that your HTTP response is slightly different from mine. 
Py default pydantic ignores any unrecognized field, that's maybe a reason why you do not see some fields.
Could you please take a look at the plain JSON response you got from oew.com?

**Comment by disforw** [comment]
Here… whats this?
![IMG_3464](https://github.com/kdeyev/eyeonwater/assets/24241104/1e5c1c1c-8fa4-4853-90dd-ec421e956808)

**Comment by kdeyev** [comment]
It seems to be a temperature in Celsius degree. It's a bit warm for water in the pipe, but Celsius is my best guess.
I have very similar values in my reading, but I don't really know the data model - I did my best to deduce it from the payload.

**Comment by disforw** [comment]
I’m going to create these as sensors in my dev environment and test my water temperature for a couple of days. It will be fun for the kids. I’ll get back to you about this…

**Comment by disforw** [comment]
Ok, I have learned quite a bit about tap water temperature over the last 24 hours! I need your help making a decision. When you have some time to read this, let me know what you want to do here.

This is going to refer back to the image I posted above…
So my water meter is located in my boiler room (as I’m sure is the same for many people). It mean’s however, that when the faucet is off, water will sit stagnant in the metal pipes and the metal meter. Because the meter is above ground and the boiler room is usually warm (or in the winter time hot) the temperature of the water sitting stagnant in the meter rises. The meter takes temperature readings all day and therefore reports a ‘latest_average’ temperature. This temperature would be INCORRECT to use as it is including readings of stagnant water!
After thinking about it though, I recommend we DO create a water temperature sensor and use the ‘seven_day_min’ reading. This number is reporting the lowest temperature that the meter has seen in the last 7 days which would essentially be the coldest water coming out of the ground….
Your thoughts?

**Comment by disforw** [comment]
And just for a laugh… this is after running water for 10 minutes (had to make sure I wasn’t testing water that was sitting in pipes)…
![IMG_3466](https://github.com/kdeyev/eyeonwater/assets/24241104/78c5479f-1705-4091-badd-80934375ac32)

**Comment by disforw** [comment]
If we move forward with this, lets use a number entity.
https://developers.home-assistant.io/docs/core/entity/number/

device_class=NumberDeviceClass.TEMPERATURE,
native_unit_of_measurement=UnitOfTemperature.CELSIUS,

**Comment by disforw** [comment]
Also, this would indicate that my meter is equipped with a temperature sensor endpoint, which may not be the case for everyone. So we need to include some logic to only create the entity if this is True.
![IMG_3468](https://github.com/kdeyev/eyeonwater/assets/24241104/b8f04125-5dc5-4cfd-af1e-132763330990)

**Comment by kdeyev** [comment]
Really great findings! 
Yes, definitely let's add seven_days_min as a temperature sensor. And as you correctly pointed the addition of the temperature sensor should be based on available sensor types. I think it will be nice to have such a sensor.

**Comment by kdeyev** [comment]
Actually, we also could do something very interesting with the 7-day average temperature. 
If we have a time series of 7-day average temperature readings - we could calculate the average temperature between readings 😄 
But it's going to be a math exercise

**Comment by disforw** [comment]
Update: I was wrong to try and use the number entity, that’s for control only. I moved the new temperature sensor in to sensor.py. That file is a mess right now with all the statistic stuff, so for now I just created a whole new sensor class. It’s using all “native” attributes so it will display in HA the way users want to see it by default.

I’d like to clean up sensor.py and move all the statistics stuff to statistics.py. Then we can move to using entity_descriptions like in binary_sensor.
![IMG_3470](https://github.com/kdeyev/eyeonwater/assets/24241104/ae3b00c9-a546-4ba3-a321-e6eaeb7e8d01)

**Comment by disforw** [comment]
The new temperature sensor is done, use HACS and redownload master to try it out. I did not put the logic that checks to see if the sensor endpoint is installed yet, so I am not releasing just yet. Looks really good.

**Comment by disforw** [comment]
This is looking good, closing…
